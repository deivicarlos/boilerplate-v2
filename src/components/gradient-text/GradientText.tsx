import { FunctionComponent } from 'react';

type Props = {
  text: string;
};

const GradientText: FunctionComponent<Props> = ({ text }) => {
  return (
    <div className="p-10 min-h-screen flex items-center justify-center bg-cool-gray-700">
      <h1 className="text-9xl font-black text-center">
        <span className="gradient-color">{text}</span>
      </h1>
    </div>
  );
};

export default GradientText;
