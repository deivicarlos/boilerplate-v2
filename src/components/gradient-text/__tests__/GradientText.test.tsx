import { render } from '@testing-library/react';
import GradientText from '../GradientText';

describe('GradientText', () => {
  const componentText = 'This is a unit test';
  let asFragment: any;
  let getByText: any;

  beforeEach(() => {
    const component = render(<GradientText text={componentText} />);

    asFragment = component.asFragment;
    getByText = component.getByText;
  });

  it('renders correctly', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct text', () => {
    expect(getByText('This is a unit test')).toBeTruthy();
  });
});
