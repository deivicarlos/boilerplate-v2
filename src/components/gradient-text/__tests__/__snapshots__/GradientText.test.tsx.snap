// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientText renders correctly 1`] = `
<DocumentFragment>
  <div
    class="p-10 min-h-screen flex items-center justify-center bg-cool-gray-700"
  >
    <h1
      class="text-9xl font-black text-center"
    >
      <span
        class="gradient-color"
      >
        This is a unit test
      </span>
    </h1>
  </div>
</DocumentFragment>
`;
